app/contexts/project_context.rb

```diff
@@ -4,11 +4,55 @@
 module ProjectContext
   extend self
 
+  REPOS_PATH = 'tmp/repos'
+
   def next_unhandled_project
     Project.unhandled.first
   end
 
   def mark_handled!(project)
     project.update!(handled_at: Time.zone.now)
   end
+
+  def clone_repo(repository_url)
+    fork(repository_url)
+    project = Project.find_by(repository_url: repository_url)
+    work_dir = directory(project)
+    return work_dir.to_s if work_dir.exist? && work_dir.directory? && !work_dir.empty?
+
+    clone(project)
+  end
+
+  def clone(project)
+    repository = Octokit::Repository.from_url(project.internal_repository_url)
+    client = Octokit::Client.new
+    repository = client.repository repository
+    git = Git.clone(repository.ssh_url, repository.name, path: directory_for_clone)
```

Comment:
context is the abstraction to make database queries, they should not work with file structure